poolapi/apijson.go

```diff
@@ -0,0 +1,41 @@
+package poolapi
+
+type Response struct {
+	Status  string      `json:"status"`
+	Message string      `json:"message"`
+	Data    interface{} `json:"data,omitempty"`
```

Comment:
and both of those are json objects with json string values.  When the json for either response is unmarshaled into this Response type it will already be unmarshaled into a map[string]string package into the empty interface.  Typing it correctly just removes the type assertion and mandates that any responses we add later follow the same format.

You could also make it a *json.RawMessage which will keep the data in a byte slice so it can be unmarshaled to the expected type.  There's no one-size-fits-all solution here and I could see clients wanting both types.